src/internal_events/journald.rs

```diff
@@ -0,0 +1,49 @@
+#![cfg(feature = "sources-journald")]
+
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub(crate) struct JournaldEventReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for JournaldEventReceived {
+    fn emit_logs(&self) {
+        trace!(message = "received line.", byte_size = %self.byte_size);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("event_processed", 1,
+                 "component_kind" => "source",
+                 "component_name" => "journald",
+        );
+        counter!("bytes_processed", self.byte_size as u64,
+                 "component_kind" => "source",
+                 "component_name" => "journald",
+        );
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct JournaldInvalidRecord {
+    pub error: serde_json::Error,
+    pub text: String,
+}
+
+impl InternalEvent for JournaldInvalidRecord {
```

Comment:
@lukesteensen you would know better. But do we want errors to end with `*Error`? So `JournaldInvalidRecordError`?

Reply:
The precedent is to not, and just make the event name obvious: `ElasticSearchMissingKeys`, `KubernetesLogsEventAnnotationFailed`, `RegexFailedMatch`, etc

Reply:
Got it. I do think we should standardize around naming rules sooner than later. I'm poking at this because I would like to derivce documentation from these structures in some way. But that's outside of the scope of this PR right now.

Reply:
I don't think it's as simple as all of these being considered errors, so the naming convention now is just precisely what happened.